packages/darklang/prettyPrinter/programTypes.dark

```diff
@@ -458,21 +455,31 @@ module Darklang =
         | EInt(_id, i) -> Int.toString i
         | EFloat(_id, sign, whole, remainder) ->
           let signPart = PACKAGE.Darklang.PrettyPrinter.sign sign
-          let wholePart = Int.toString whole
-
+          let rem = $".{remainder}" |> Float.parse
           let remainderPart =
-            match remainder with
-            | Nothing -> ""
-            | Just r -> $".{Int.toString r}"
+              match rem with
+              | Ok r ->
+                  if r == 0.0 then ""
+                  else (r  |> Float.toString) |> String.dropFirst 2
+              | Error e -> e
+          let wholePart =
+              match whole with
+              | "0" -> if remainderPart != "" then "0" else ""
+              | _ -> whole
+          let formattedFloat =
+              if remainderPart == "" then $"{wholePart}.0"
+              else $"{wholePart}.{remainderPart}"
+          $"{signPart}{formattedFloat}"
 
-          $"{signPart}{wholePart}{remainderPart}"
 
         | EChar(_id, c) -> "\"" ++ c ++ "\""
 
         | EString(_id, segments) ->
           match segments with
           | [] -> "\"\""
-          | [ StringText text ] -> $"\"{text}\""
+          | [ StringText text ] ->
+            let escapedText = String.replaceAll text "\n" "\\n"
```

Comment:
I used replaceAll in this case since I couldn't use String.map  (wasn't allowed to replace a char with a string). 
Do you have any suggestions for a more efficient approach that would allow checking for things other than line breaks, such as spaces?

Reply:
What about something like:

`String.split ... |> List.map (replace ...) |> String.concat`

Reply:
That said, `String.replaceAll` might be fine. You could do something like:

```
text
|> String.replaceAll "\n" "\\n"
|> String.replaceAll "\t" "\\t"
|> String.replaceAll "\r" "\\r"
```